release/pkg/manager/calico/manager.go

```diff
@@ -568,36 +626,19 @@ func (r *CalicoManager) hashreleasePrereqs() error {
 			return fmt.Errorf("missing hashrelease server configuration")
 		}
 	}
+
 	if r.publishImages {
 		return r.assertImageVersions()
 	} else {
-		results := make(map[string]imageExistsResult, len(r.imageComponents))
-		ch := make(chan imageExistsResult)
-		for name, component := range r.imageComponents {
-			go imgExists(name, component, ch)
-		}
-		for range images {
```

Comment:
I think this should be `for range r.imageComponents` - the thing is that we're waiting for each started goroutine to finish, and we know we launched `r.imageComponents` goroutines. That would probably fix the original deadlock IIUC